sanitizers/detectors/detector.go

```diff
@@ -1,9 +1,103 @@
 package detectors
 
-// Detector defines the interface for the bug detectors.
+import (
+	"os/exec"
+	"text/template/parse"
+)
+
+// Define an enum variant, for each implemented `Detector`
+type Detectors int64
+
+const (
+	SQLInjection Detectors = iota
+	TemplateInjection
+	PathTraversal
+	CommandInjection
+)
+
+var SQLIDet Detectors = SQLInjection
+var TIDet Detectors = TemplateInjection
+var PTDet Detectors = PathTraversal
+var CIDet Detectors = CommandInjection
+
+func (d Detectors) String() string {
+	return []string{"SQLInjection", "TemplateInjection", "PathTraversal", "CommandInjection"}[d]
+}
+
+// This is the parent DetectorClass als childs inherit from
+// It's a place to define common properties that all detectors should share/need
+type DetectorClass struct {
```

Comment:
Could you explain the reasoning behind having a `DetectorClass` struct that aggregates all the fields needed by all detectors and a dedicated struct with the relevant fields for each detector?